### ISSUE TITLE
POPCNT

### ISSUE THREAD

**Issue by steakknife** [steakknife/hamming#2]
http://danluu.com/assembly-intrinsics/

**Comment by steakknife** [comment]
`227 ns/op` :(

```Go
/*
#include <assert.h>
#include <stdbool.h>
#include <stdint.h>
#include <stdio.h>

#if defined(__x86_64__) || defined(__x86_64) || \
    defined(__amd64__) || defined(__amd64) || \
    defined(amd64) || defined(_M_AMD64) || defined(_M_X64)
# define X86_64
#endif


static inline bool
detect_pop_cnt(void) {
#if defined(X86_64)
   uint64_t features;
   __asm__ __volatile__ (
    "cpuid\n"
    : "=c" (features)
    : "a"(0x00000001)
    : "%rbx", "%rdx"
  );
  printf("features: 0x%02llX\n", features);
  // Advanced Bit Manipulation Bit (ABM)
  return (((features >> 23) & 1) ? true : false);
#else
  printf("features: not 64bit\n");
  return (false);
#endif
}

static inline int
popcnt(uint64_t x) {
#if defined(X86_64)
   register int out;
   __asm__ (
    "popcnt\t%%rdx,%%rcx\n"
    : "=c" (out)
    : "c" (x), "d" (x)
  );
  return ((int)out);
#else
  assert(0);
  return (0);
#endif
}
*/
import "C"
```

**Comment by steakknife** [comment]
The issue is the jumping around & stack buildup/teardown from Go -> C -> Go is slow.

**Comment by steakknife** [comment]
Added C-less assembly as of 0.1.0 for amd64

Calling cpuid would be a good idea

**Comment by steakknife** [comment]
Resolved in 0.1.1
